### PR TITLE
DOP-4248: call custom github action to store to S3

### PR DESCRIPTION
**Pull request by seungpark**
### Stories/Links:

DOP-4248

### Staging Links:

[Sample content uploaded to S3 bucket](https://docs-mongodb-org-dotcomstg.s3.us-east-2.amazonaws.com/gh-actions/cloud-docs/ce17ed5b91c514e31c992b35df25cb015bb3f1de/index.html)
S3 contents can be explored at [this link](https://s3.console.aws.amazon.com/s3/buckets/docs-mongodb-org-dotcomstg?region=us-east-2&bucketType=general&prefix=gh-actions/&showversions=false)

### Notes:
- This is accompanied by the [worker actions PR](https://github.com/mongodb/docs-worker-actions/pull/79) that takes the contents of an input directory and uploads to S3.
- A new IAM user `docs-snooty-ci` has been generated and the access key + secret access key have been added to repo secrets.
- Log of `Publish build` step can be viewed [here](https://github.com/mongodb/snooty/actions/runs/7836852714/job/21385329908?pr=1003)

### CODE REVIEW

**Comment by mmeigs** [comment]
This looks good in all ways except when I visit the newly hosted pages. I don't know where this is happening. Probably somewhere in the build or maybe S3 dynamics... but the uploaded "sites" are failing on all their fetches with an error `"invalid session: access token expired"`

**Comment by seungpark** [comment]
> This looks good in all ways except when I visit the newly hosted pages. I don't know where this is happening. Probably somewhere in the build or maybe S3 dynamics... but the uploaded "sites" are failing on all their fetches with an error `"invalid session: access token expired"`

discussed offline. seems this is irreproducible, possibly a machine related error (?)

File: .github/workflows/stage-docs-site.yml [hunk 1]
```diff
@@ -29,9 +31,25 @@ jobs:
...
           npm ci --legacy-peer-deps
           npm run build
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-2
+      - name: Publish build
+        uses: mongodb/docs-worker-actions/publish-build@DOP-4248
```
Comment (mmeigs) [inline .github/workflows/stage-docs-site.yml:45]: just a reminder to change to @main when the Action is merged

**Review by mmeigs — APPROVED** [review]
Looks great. Approving now, but I realize you have to merge the Action first!

File: .github/workflows/stage-docs-site.yml [hunk 2]
```diff
@@ -29,9 +31,25 @@ jobs:
...
+        env:
+          AWS_BUCKET: docs-mongodb-org-dotcomstg
+          SOURCE_DIR: public
+          DESTINATION_DIR: ${{ env.s3_root_dir }}
+          COMMIT_HASH: ${{ github.sha }}
```
Comment (branberry) [inline .github/workflows/stage-docs-site.yml:50]: NIT: I believe you should be able to access `github` context values from within the custom action itself by importing `@actions/github`, but this is also totally fine as is.

**Review by branberry — APPROVED** [review]

File: .github/workflows/stage-docs-site.yml (hunk 1 above)
Comment (seungpark) [reply .github/workflows/stage-docs-site.yml:45]: thanks for the reminder! almost forgot post vacation :D

**Pull request merged**